src/Stan/Inspection/Partial.hs

```diff
@@ -43,23 +59,67 @@ mkPartialInspection insId NameMeta{..} = Inspection
     , inspectionSeverity = Severe
     }
 
--- | 'Id' for the partial 'head' 'Inspection' — @STAN-0001@.
-stan0001 :: Id Inspection
-stan0001 = Id "STAN-0001"
-
--- | Corresponding 'Inspection' for 'stan0001' — partial 'GHC.List.head' @STAN-0001@.
-stan0001Inspection :: Inspection
-stan0001Inspection = (mkPartialInspection stan0001 stan0001Meta)
+{- | Smart constructor to create partial 'Inspection' for functions
+that work with lists.
+-}
+mkPartialInspectionList :: Id Inspection -> NameMeta -> Inspection
+mkPartialInspectionList insId nameMeta = (mkPartialInspection insId nameMeta)
     { inspectionSolution =
         [ "Replace list with 'NonEmpty' from 'Data.List.NonEmpty'"
         , "Use explicit pattern-matching over lists"
         ]
     }
 
--- | Corresponding 'NameMeta' for 'stan0001' — partial 'GHC.List.head' @STAN-0001@.
-stan0001Meta :: NameMeta
-stan0001Meta = NameMeta
-    { nameMetaName       = "head"
+mkBaseListMeta :: Text -> NameMeta
+mkBaseListMeta funName = NameMeta
+    { nameMetaName       = funName
     , nameMetaPackage    = "base"
     , nameMetaModuleName = "GHC.List"
     }
```

Comment:
I will need this function for infinite checks as well... Should we create `Stan.NameMeta` or should we move into `Stan.Inspections.Common` or something else?

Reply:
`Stan.NameMeta` sounds like the right place. I can do this :+1: